packages/account/src/Components/api-token/__tests__/api-token-delete-button.spec.tsx

```diff
@@ -0,0 +1,104 @@
+        modal_root_el.setAttribute('id', 'modal_root');
+        document.body.appendChild(modal_root_el);
+    });
+
+    afterAll(() => {
+        document.body.removeChild(modal_root_el);
+    });
+
+    it('should render ApiTokenDeleteButton', () => {
+        renderAPIDeleteButton();
+        expect(screen.getByTestId('dt_token_delete_icon')).toBeInTheDocument();
+        expect(screen.queryByText('Delete this token')).not.toBeInTheDocument();
+    });
+
+    it('should display Delete this token when mouse enter', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.hover(delete_icon);
+        expect(screen.getByText('Delete this token')).toBeInTheDocument();
+    });
+
+    it('should not display Delete this token when mouse leave', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.hover(delete_icon);
+        expect(screen.getByText('Delete this token')).toBeInTheDocument();
+        userEvent.unhover(delete_icon);
+        expect(screen.queryByText('Delete this token')).not.toBeInTheDocument();
+    });
+
+    it('should display Popup when delete icon is clicked', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByText('Delete token')).toBeInTheDocument();
+        expect(screen.getByText('Are you sure you want to delete this token?')).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: 'Yes, delete' })).toBeInTheDocument();
+    });
+
+    it('should close the modal when clicked on Cancel', async () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        const cancel_button = screen.getByRole('button', { name: 'Cancel' });
+        userEvent.click(cancel_button);
+        await waitFor(() => expect(screen.queryByText('Delete token')).not.toBeInTheDocument());
+        expect(screen.queryByText('Are you sure you want to delete this token?')).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Cancel' })).not.toBeInTheDocument();
+        expect(screen.queryByRole('button', { name: 'Yes, delete' })).not.toBeInTheDocument();
+    });
+
+    it('should should trigger deleteToken when clicked on Yes, delete', () => {
+        renderAPIDeleteButton();
+        const delete_icon = screen.getByTestId('dt_token_delete_icon');
+        userEvent.click(delete_icon);
+        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+        const delete_token_button = screen.getByRole('button', { name: 'Yes, delete' });
```

Comment:
recommendation for maintainability: to introduce variables for repeated strings.
>const deletion_confirmation = 'Yes, delete';
//... name: deletion_confirmation